bindings/cpp/session.cpp

```diff
@@ -2249,7 +2264,46 @@ async_iterator_result session::start_iterator(const key &id, const std::vector<d
 	req->time_end = time_end;
 	req->range_num = ranges.size();
 
-	memcpy(data.skip<dnet_iterator_request>().data(), &ranges.front(), ranges_size);
+	if (ranges_size)
+		memcpy(data.skip<dnet_iterator_request>().data(), &ranges.front(), ranges_size);
+
+	return iterator(id, data);
+}
+
+async_iterator_result session::start_copy_iterator(const key &id,
+		const std::vector<dnet_iterator_range>& ranges,
+		uint64_t flags,
```

Comment:
`DNET_IFLAGS_NO_META` for this iterator will lead to copying / moving data without meta - it may be kinda strange behaviour, but it is ok and should be mentioned in docs.